src/ray/common/scheduling/cluster_resource_scheduler.h

```diff
@@ -343,25 +356,46 @@ class ClusterResourceScheduler {
   ///
   /// \param available A list of available capacities for resource's instances.
   /// \param resource_instances List of the resource instances being updated.
-  void AddAvailableResourceInstances(std::vector<double> available,
-                                     ResourceInstanceCapacities *resource_instances);
+  ///
+  /// \return Overflow capacities of "resource_instances" after adding instance 
+  /// capacities in "available". 
+  std::vector<double> AddAvailableResourceInstances(std::vector<double> available,
+                                                    ResourceInstanceCapacities *resource_instances);
 
   /// Decrease the available capacities of the instances of a given resource.
   ///
   /// \param free A list of capacities for resource's instances to be freed.
   /// \param resource_instances List of the resource instances being updated.
-  void SubtractAvailableResourceInstances(std::vector<double> free,
-                                          ResourceInstanceCapacities *resource_instances);
+  /// \return Under capacities of "resource_instances" after substracting instance 
```

Comment:
Same comment here, can you explain what under capacities means? Giving a formula would be helpful.

Reply:
done